### ISSUE TITLE
[Accessibility]: Focus on Implementations in ref panel after clicking Find implementations

### ISSUE THREAD

**Issue by varungandhi-src** [sourcegraph/sourcegraph#36590]
### Audit type

Screen reader navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/33511

### Problem description

On clicking Find implementations in the code intel popover, the focus still stays on the button.

### Expected behavior

The focus should go to the ref panel after clicking the Find implementations, maybe specifically to the Implementations section of the ref panel.

### Additional details

Related:
- https://github.com/sourcegraph/sourcegraph/issues/36568

### Assigning labels

- [ ] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.

**Comment by umpox** [comment]
Might be fixed by https://github.com/sourcegraph/sourcegraph/issues/36581
